data_structures/src/radon_report.rs

```diff
@@ -30,13 +32,33 @@ where
     RT: TypeLike,
 {
     /// Factory for constructing a `RadonReport` from the `Result` of something that could be
-    /// `ErrorLike` plus a `ReportContext`.
+    /// `TypeLike` or `ErrorLike` plus a `ReportContext`.
     pub fn from_result(result: Result<RT, RT::Error>, context: &ReportContext) -> Self {
         let intercepted = RT::intercept(result);
-
         RadonReport {
+            metadata: context.stage.clone(),
+            partial_results: None,
             result: intercepted,
+            running_time: context.duration(),
+        }
+    }
+
+    /// Factory for constructing a `RadonReport` from a vector of partial results, which could be
+    /// `TypeLike` or `ErrorLike`, plus a `ReportContext`.
+    pub fn from_partial_results(
+        partial_results: Vec<Result<RT, RT::Error>>,
+        context: &ReportContext,
+    ) -> Self {
+        let intercepted: Vec<RT> = partial_results.into_iter().map(RT::intercept).collect();
+        let result = match intercepted.last() {
+            None => unreachable!("Partial result vectors always contain at least 1 item"),
```

Comment:
I can trigger this branch using `RadonReport::from_partial_results(vec![], context)`, therefore it is not unreachable. 

```suggestion
            None => panic!("Partial result vectors must always contain at least 1 item"),
```